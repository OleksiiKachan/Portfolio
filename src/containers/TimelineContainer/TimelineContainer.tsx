import React, { Component } from 'react';
import TimelineSection from '../../components/TimelineSection';

/*
    <TimelineContainer projects={} />
*/
export default class ProjectsContainer extends Component<any, any> {
  render() {
    const dataStore = require('../../data.json');
    return (
      <TimelineSection
        experience={dataStore.experience}
        education={dataStore.education}
      />
    );
  }
}
