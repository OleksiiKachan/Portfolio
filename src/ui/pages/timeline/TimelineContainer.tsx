import React, { Component } from 'react';
import TimelineSection from './TimelineSection/TimelineSection';

/*
    <TimelineContainer />
*/

export default class ProjectsContainer extends Component<{}> {
  render() {
    const dataStore = require('../../../data.json');
    return (
      <TimelineSection
        experience={dataStore.experience}
        education={dataStore.education}
      />
    );
  }
}
