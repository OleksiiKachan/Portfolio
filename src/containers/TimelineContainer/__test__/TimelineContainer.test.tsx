import React from 'react';
import { shallow } from 'enzyme';
import TimelineContainer from '..';

describe('TimelineContainer', () => {
  it('Snapshot', () => {
    const component = shallow(<TimelineContainer />);
    expect(component).toMatchSnapshot();
  });
});
