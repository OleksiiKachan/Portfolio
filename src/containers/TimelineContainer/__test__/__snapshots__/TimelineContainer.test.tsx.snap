// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineContainer Snapshot 1`] = `
<_default
  education={
    Array [
      Object {
        "dates": "2013-2017",
        "degree": "Bachelor of Software Engineering",
        "department": "Faculty of Informatics and Computer Science, Department of Technical Cybernetics",
        "id": "kpi",
        "program": "Software Engineering",
        "school": "National Technical University of Ukraine \\"Kyiv Polytechnic Institute\\"",
      },
      Object {
        "dates": "2017-2019",
        "degree": "Ontario Graduate Certificate",
        "department": "School of Media Studies & Information Technology",
        "id": "humber",
        "program": "Information Technology Solutions",
        "school": "The Humber Institute of Technology and Advanced Learning",
      },
    ]
  }
  experience={
    Array [
      Object {
        "companyTitle": "Ltd. \\"Kontynium-Treyd\\"",
        "dates": "2014",
        "id": "kontynium-treyd",
        "position": "Technician Programmer of administrative and economic department",
        "responsibilities": Array [
          "System administration of the enterprise using 1C: Enterprise (ERP-like platform)",
          "Maintenance of computers, peripheral and specialized equipment (common printers, Zebra printers, electronic scales, fiscal cash registers, etc.)",
        ],
      },
      Object {
        "companyTitle": "rLoop Incorporated",
        "dates": "2016-2017",
        "id": "rloop",
        "position": "Software and Systems Engineer",
        "projects": Array [
          Object {
            "id": "rpod",
            "title": "Hyperloop rPod",
          },
        ],
        "responsibilities": Array [
          "Software & System engineering",
          "UI & UX Design & Development",
        ],
      },
      Object {
        "companyTitle": "OLSOM L.L.C",
        "dates": "2017",
        "id": "olsom",
        "position": "Software Engineer",
        "projects": Array [
          Object {
            "id": "agw",
            "title": "AGW modular complex solution for automotive industry",
          },
          Object {
            "id": "ftp-service",
            "title": "FTP Transfer Service",
          },
          Object {
            "id": "tdm",
            "title": "TDM (Tools Data Manager) for automotive industry",
          },
          Object {
            "id": "cardboard-prod",
            "title": "Corrugated Cardboard Production System",
          },
        ],
        "responsibilities": Array [
          "Development and support of the company's products: AGW modular complex solution for automotive industry; TDM (Tools Data Manager) for automotive industry",
          "Development and support of the client projects based on company's products (1-3-tier automotive manufacturers (Volkswagen, Opel, Tesla, etc.)",
          "Development of outsourced minor projects",
          "Development of internal tools",
        ],
      },
    ]
  }
/>
`;
